### PR TITLE
 rac-3789 run_last.d symlink should be relative rather than absolute

### PR DESCRIPTION
**Pull request by stuart-stanley**
Primary changes:
* changed abs symlink to be a link within the same dir
* added logger_get_logging_dir to infra-logging (for test use)

TDD changes:
* fixed 'collect_only' check so it works when testing the plugins as well as when the plugin is used!
* added test_infra_logging.py as location for tests of the infra-logging stuff.
** added setup to ask infra-logging (via the logger_get_logging_dir from above) where to look
** added canary test to make sure tests are being invoked
** added test to check the run_last.d is there, is a symlink, and points to something valid
** added fail-before-change, pass-after-change test to make sure the symlink doesn't start with a '/'. I.E., it's relative.
* changed test_stream_base to use nose-created singleton for StreamMonitorPlugin instance.

Nearby-changes:
* change both test and test/stream-monitor's mkenv.sh's to upgrade the venv pip. This speeds up generation of the virtual-env's signifigently and appears to finally get rid of errors based on trying to compile things.
* removed wheel pinned to 0.24.0 from requirements.txt (this is taken care of automagically)
* removed bogus __name__ == '__main__' code from infra_logging.py

@RackHD/corecommitters @johren @hohene @gpaulos @larry-dean @jimturnquist @keedya @panpan0000 @changev @anhou 

(note: reviewer list built up from "people I think _might_ be interested in this" ;) )
This is NOT needed for phase-1, but should not hurt anything. It also paves the way to be able to test rac-3817 (getting captured data back to pre-pull-544 levels)

### CODE REVIEW

File: test/stream-monitor/sm_plugin/stream_monitor.py [hunk 1]
```diff
@@ -45,7 +51,7 @@ def configure(self, options, conf):
         super(StreamMonitorPlugin, self).configure(options,conf)
         if not self.enabled:
             return
-        if conf.options.collect_only:
+        if getattr(conf.options, 'collect_only', False):
```
Comment (hohene) [inline test/stream-monitor/sm_plugin/stream_monitor.py]: With this change, should I not see any logging info when I run_tests.py with -list option?  This is current output.
```
(eh-stu-ub16) hohene@lab17:~/sandbox-github/rackhd/stu-rackhd/test$ run_tests.py -config config-mn -stack 1 -test tests/compute/test_rackhd11_computenode_pollers.py -list
2017-01-04 15:59:53,419 22684 MainProcess stream-monitor/flogging/infra_logging.py:__init__@105 gl-main INFO this runs logging dir /emc/hohene/sandbox-github/rackhd/stu-rackhd/test/log_output/run_2017-01-04_15:59:53.d

Test Listing for: tests/compute/test_rackhd11_computenode_pollers.py
----------------------------------------------------------------------
2017-01-04 15:59:53,934 22691 MainProcess stream-monitor/flogging/infra_logging.py:__init__@105 gl-main INFO removing previous logging dir /emc/hohene/sandbox-github/rackhd/stu-rackhd/test/log_output/run_2017-01-04_15:43:52.d
2017-01-04 15:59:53,935 22691 MainProcess stream-monitor/flogging/infra_logging.py:__init__@105 gl-main INFO this runs logging dir /emc/hohene/sandbox-github/rackhd/stu-rackhd/test/log_output/run_2017-01-04_15:59:53.d
**** Using config file path: config-mn
test_1_verify_pollers (test_rackhd11_computenode_pollers.rackhd11_computenode_pollers) ... ok
test_2_pollers_by_id (test_rackhd11_computenode_pollers.rackhd11_computenode_pollers) ... ok
test_3_poller_headers (test_rackhd11_computenode_pollers.rackhd11_computenode_pollers) ... ok
test_4_poller_default_cache (test_rackhd11_computenode_pollers.rackhd11_computenode_pollers) ... ok
test_5_poller_current_data (test_rackhd11_computenode_pollers.rackhd11_computenode_pollers) ... ok
test_6_poller_status_timestamp (test_rackhd11_computenode_pollers.rackhd11_computenode_pollers) ... ok
test_7_nodes_id_pollers (test_rackhd11_computenode_pollers.rackhd11_computenode_pollers) ... ok

----------------------------------------------------------------------
Ran 7 tests in 0.005s
```

Comment (stuart-stanley) [inline test/stream-monitor/sm_plugin/stream_monitor.py]: You won't see logging from within the tests as they are probed. 
Note that this exact change (using getattr) leaves the behavior the same for "normal" tests, but allows the plugin self-tests to be run. 

Would love to totally kill all logging (and log directory rotation) for -list, but that is a much much (MUCH!) trickier thing to solve.

**Comment by hohene** [comment]
Ran test suite with latest updates.  +1

**Comment by gpaulos** [comment]
Also ran suite with latest updates. +1

**Comment by stuart-stanley** [comment]
ug. that did not go right (trying to rebase). Please don't merge until I untangle it. sigh.

**Pull request closed without merging**